src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java

```diff
@@ -605,19 +605,22 @@ public Object run() {
             }
         });
 
-        // Clone a new stack not interfering with the current stack.
-        AuthorizationStack newStack = getStack().clone();
-
+        AuthorizationStack newLocalStack = this.newStack;
+        if (newLocalStack == null) {
+            // Clone a new stack not interfering with the current stack.
+            newLocalStack = getStack().clone();
+        }
```

Comment:
I'd like to see also `this.newStack.clone()` so that noone from the outside can modify the stack.

Reply:
fixed